File: go-client-sdk/pkg/models/operations/getonecrmuser.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetOneCrmUserQueryParams struct {
	AccessToken string `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool  `queryParam:"style=form,explode=true,name=allFields"`
	ID          string `queryParam:"style=form,explode=true,name=id"`
}

type GetOneCrmUserResponseBody struct {
	User *shared.User `json:"user,omitempty"`
}

type GetOneCrmUserRequest struct {
	QueryParams GetOneCrmUserQueryParams
}

type GetOneCrmUserResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetOneCrmUserResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/getonecrmuser.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetOneCrmUserQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=id" })
  id: string;
}


export class GetOneCrmUserResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=user" })
  user?: shared.User;
}


export class GetOneCrmUserRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetOneCrmUserQueryParams;
}


export class GetOneCrmUserResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetOneCrmUserResponseBody;
}
